happybase/connection.py

```diff
@@ -150,7 +163,20 @@ def _refresh_thrift_client(self):
         if self.timeout is not None:
             socket.setTimeout(self.timeout)
 
-        self.transport = self._transport_class(socket)
+        if not self._use_kerberos:
+            self.transport = self._transport_class(socket)
+        else:
+            # Check for required arguments for kerberos
+            if self._transport_class is not TBufferedTransport:
+                raise ValueError("Must use a buffered transport "
+                                 " when use_kerberos is enabled")
+
+            saslc = sasl.Client()
+            saslc.setAttr("host", self.host)
+            saslc.setAttr("service", self._sasl_service)
+            saslc.init()
+            self.transport = TSaslClientTransport(saslc, "GSSAPI", socket)
```

Comment:
hmmm... would it make sense to have a `sasl` transport instead of a `use_kerberos` arg? not sure, i'm not familiar with this.